### PR TITLE
Add passwordless test helpers

### LINKED ISSUE #71: Test helpers
Something like Devise's `sign_in @user` helper.

See also #29.

### PR DESCRIPTION
**Pull request by rickychilcott**
This is a PR to add test helpers for test unit and rspec. I tried this locally on a project I'm using and it works ok, but I'm only using rspec. Someone who has a good example test unit project should try this or some tests can be integrated into a passwordless example app.

This takes a lot of code and ideas from #71 so no credit is taken here, just trying to pull together to get to this as part of a 1.0 release (#89).

Close #71 and address #29

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Add authentication to your Rails app without all the icky-ness of passwords.
   * [Token and Session Expiry](#token-and-session-expiry)
   * [Redirecting back after sign-in](#redirecting-back-after-sign-in)
   * [Claiming tokens](#claiming-tokens)
+* [Testing](#testing)
```
Comment (mikker) [inline README.md:29]:
```suggestion
* [Test helpers](#testing)
```

File: README.md [hunk 2]
```diff
@@ -340,6 +341,32 @@ end
 ```
 </details>
 
+## Testing
```
Comment (mikker) [inline README.md:344]:
```suggestion
## Test helpers
```

File: README.md [hunk 3]
```diff
@@ -340,6 +341,32 @@ end
...
+
+To test controller that are protected by a `before_action :require_user`, require Passwordless's test helpers in your test suite.
```
Comment (mikker) [inline README.md:346]:
```suggestion
To help with testing, a set of test helpers are provided.
```

File: README.md [hunk 4]
```diff
@@ -340,6 +341,32 @@ end
...
+
+For `rspec`, add the following line to your `spec/rails_helper.rb` or
```
Comment (mikker) [inline README.md:348]:
```suggestion
If you are using RSpec, add the following line to your `spec/rails_helper.rb` or
```

File: README.md [hunk 5]
```diff
@@ -340,6 +341,32 @@ end
...
+`spec/spec_helper` if `rails_helper` does not exist:
```
Comment (mikker) [inline README.md:349]:
```suggestion
`spec/spec_helper.rb` if `rails_helper` does not exist:
```

File: README.md [hunk 6]
```diff
@@ -340,6 +341,32 @@ end
...
+```ruby
+require "passwordless/test_helpers"
+```
+
+
+Then in your controller, request, and system tests/specs, you can utilize the following methods:
+
+```ruby
+passwordless_sign_in # takes you to the sign in path
+passwordless_sign_in(user) # signs you in as a user
```
Comment (mikker) [inline README.md:366]: I'm not sure I like this dual meaning API. Let's split it into `passwordless_sign_in(resource)` and `passwordless_sign_in_path`.

**Review by mikker — CHANGES_REQUESTED** [review]
Thank you for this @rickychilcott!

They'll have to be modified a bit for 1.0 but that doesn't mean they aren't a great addition. A few remarks and suggestions, otherwise looks great!

**Comment by rickychilcott** [comment]
Thanks @mikker. I made the requested changes.

I was modeling some of the syntax off of https://github.com/thoughtbot/clearance#controller-test-helpers but I looked too quickly and didn't realize that they effectively shim the `current_user` if you just did a `sign_in` without doing `sign_in_as(user)`, so I dropped my weird overloading of the `passwordless_sign_in`/`passwordless_sign_in(user)` form. 

Let me know if you'd like any other changes.

**Review by mikker — APPROVED** [review]
💙💚💛💜❤️

**Pull request merged**